src/main/java/de/unistuttgart/iste/gits/reward/service/RewardService.java

```diff
@@ -0,0 +1,146 @@
+package de.unistuttgart.iste.gits.reward.service;
+
+import de.unistuttgart.iste.gits.common.event.UserProgressLogEvent;
+import de.unistuttgart.iste.gits.generated.dto.Content;
+import de.unistuttgart.iste.gits.generated.dto.RewardScores;
+import de.unistuttgart.iste.gits.reward.persistence.dao.AllRewardScoresEntity;
+import de.unistuttgart.iste.gits.reward.persistence.dao.RewardScoreEntity;
+import de.unistuttgart.iste.gits.reward.persistence.mapper.RewardScoreMapper;
+import de.unistuttgart.iste.gits.reward.persistence.repository.AllRewardScoresRepository;
+import de.unistuttgart.iste.gits.reward.service.calculation.*;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+@Transactional
+public class RewardService {
+
+    private final AllRewardScoresRepository rewardScoresRepository;
+    private final RewardScoreMapper mapper;
+    private final CourseServiceClient courseServiceClient;
+    private final ContentServiceClient contentServiceClient;
+
+    /**
+     * Recalculates the reward scores for a given user and course.
+     *
+     * @param courseId the id of the course
+     * @param userId   the id of the user
+     * @return the recalculated reward scores
+     */
+    public RewardScores recalculateScores(UUID courseId, UUID userId) {
+        AllRewardScoresEntity allRewardScoresEntity = rewardScoresRepository
+                .findById(new AllRewardScoresEntity.PrimaryKey(courseId, userId))
+                .orElseGet(() -> initializeRewardScores(courseId, userId));
+
+        List<UUID> chapterIds = courseServiceClient.getChapterIds(courseId);
+
+        List<Content> contents = contentServiceClient.getContentsWithUserProgressData(userId, chapterIds);
+
+        allRewardScoresEntity
+                .setHealth(new HealthScoreCalculator().recalculateScore(allRewardScoresEntity.getHealth(), contents));
```

Comment:
This is ugly. IMO, as the ScoreCalculator interface is never used, it's unnecessary. Why not just use static methods?

Or if you really want that interface, use singletons.

Reply:
I can add them as constructor fields if you like, static methods are pretty ugly too, e.g. if we want to mock that stuff or want to have different calculation methods in the future or whatever. 
The interface is unnecessary but also it kinda makes sense that all of them work the same so I thought an interface makes sense

Reply:
And why not privatize the constructor and provide a getInstance() singleton method? Then we can still have the interface in case we want to pass around calculators in methods later on and we can also mock the stuff.

Reply:
Done as discussed in discord